lib/ramble/ramble/renderer.py

```diff
@@ -67,8 +71,44 @@ def render_objects(self, variables, matrices=None):
             object_variables[name] = value
 
         new_objects = []
+        defined_zips = {}
+        consumed_zips = set()
         matrix_objects = []
 
+        if zips:
+            zipped_vars = set()
+
+            for zip_group, group_def in zips.items():
+                # Create a new defined zip
+                defined_zips[zip_group] = {'vars': {}, 'length': -1}
```

Comment:
Is `-1` actually important here, over say using 0? 

I see later you check if it's -1, and are using it as a "I set this value not someone else", but I think 0 would be the same and more truthy?

Reply:
Well, I wanted to support:

`var_name: []` which was why I used `-1`. But maybe I should error if a vector is passed with a length of zero, since we don't have values to extract for the variable.